bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/ItemStateTriggerHandler.java

```diff
@@ -42,6 +42,7 @@
 
     private String itemName;
     private String state;
+    private String oldState;
```

Comment:
Yes, it already handles change events and it does so without the need of this field, so please remove it.